app/src/main/java/org/simple/clinic/home/overdue/OverdueListAdapter.kt

```diff
@@ -18,11 +19,41 @@ import org.simple.clinic.patient.Gender
 import org.simple.clinic.util.Truss
 import java.util.UUID
 
-class OverdueListAdapter(val phoneCallClickStream: PublishSubject<CallPatientClicked>) : ListAdapter<OverdueListItem, OverdueListViewHolder>(OverdueListDiffer()) {
+class OverdueListAdapter(private val phoneCallClickStream: PublishSubject<CallPatientClicked>) : ListAdapter<OverdueListItem, OverdueListViewHolder>(OverdueListDiffer()) {
+
+  private lateinit var recyclerView: RecyclerView
+
+  override fun onAttachedToRecyclerView(rv: RecyclerView) {
+    super.onAttachedToRecyclerView(rv)
+    recyclerView = rv
+  }
 
   override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): OverdueListViewHolder {
     val layout = LayoutInflater.from(parent.context).inflate(R.layout.item_overdue_list, parent, false)
-    return OverdueListViewHolder(layout, phoneCallClickStream)
+    val holder = OverdueListViewHolder(layout, phoneCallClickStream)
+
+    layout.setOnClickListener {
+      holder.handleBottomLayoutVisibility()
+
+      holder.itemView.post {
+        val itemLocation = IntArray(2)
+        holder.itemView.getLocationOnScreen(itemLocation)
+
+        val rvLocation = IntArray(2)
+        recyclerView.getLocationOnScreen(rvLocation)
+
+        val rvBottom = rvLocation[1] + recyclerView.height
+        val itemBottomMargin = (holder.itemView.layoutParams as ViewGroup.MarginLayoutParams).bottomMargin
+        val itemBottom = itemLocation[1] + holder.itemView.height + itemBottomMargin
+
+        val differenceInBottoms = itemBottom - rvBottom
+        if (differenceInBottoms > 0) {
+          (holder.itemView.parent as RecyclerView).smoothScrollBy(0, differenceInBottoms)
+        }
+      }
```

Comment:
Can we move this logic to a different place? A function in this adapter is also fine.

Reply:
Or an extension function

Reply:
I was thinking this could be refactored into something like this to make it more readable. Right now it looks like gibberish because we're accessing arrays using indices instead of typed names.

```kotlin
holder.itemView.post {
  val itemLocation = holder.itemView.locationRectOnScreen()
  val itemBottomWithMargin = itemLocation.bottom + holder.itemView.marginLayoutParams.bottomMargin

  val rvLocation = recyclerView.locationRectOnScreen()
  val differenceInBottoms = itemBottomWithMargin - rvLocation.bottom

  if (differenceInBottoms > 0) {
    (holder.itemView.parent as RecyclerView).smoothScrollBy(0, differenceInBottoms)
  }
}
```

Extension functions in `Views.kt`:

```kotlin
fun View.locationRectOnScreen(): Rect {
  val location = IntArray(2)
  getLocationOnScreen(location)
  return Rect(location[0], location[1], location[0] + width, location[1] + height)
}

val View.marginLayoutParams: ViewGroup.MarginLayoutParams
  get() = layoutParams as ViewGroup.MarginLayoutParams
```